source/controllers/include/controllers/impedance/Dissipative.hpp

```diff
@@ -303,14 +303,6 @@ void Dissipative<S>::compute_damping(const Eigen::VectorXd& desired_velocity) {
   this->set_damping(this->basis_ * this->get_diagonal_eigenvalues() * this->basis_.transpose());
 }
 
-template<class S>
-S Dissipative<S>::compute_command(const S& desired_state, const S& feedback_state) {
```

Comment:
I'm tempted to suggest templating the `compute_damping<S>` (or even some helper function `auto get_state_data<S>(S desired state)`) and do the necessary specialisations there. That way you can leave this `compute_command` templated without specialisation, and you don't need to call `this->Impedance<S>::compute_command` repeatedly.

Reply:
Yes that is a good idea. Let me give it a try.

Reply:
I made a few changes that I hope go in the direction you suggested. I also simplified a lot the unit tests.